### ISSUE TITLE
Error updating assistant in place

### ISSUE THREAD

**Issue by nathan-gilbert** [skyscrapr/terraform-provider-openai#78]
Trying the new 1.2.1 version of the provider and am seeing this error message when updating in place. This assistant was created with the previous version of this provider if that matters.

<img width="1038" alt="image" src="https://github.com/skyscrapr/terraform-provider-openai/assets/546222/d16a28cd-ea7b-42f5-b17f-428e90c47244">

The biggest change to the code was swapping `file_ids` for the new `tool_resources` attributes object.

**Comment by nathan-gilbert** [comment]
My resource looks like:

```terraform

tool_resources = {
    file_search = {
      vector_stores = {
        file_ids = [
          openai_file.testeract-baseline-file-1.id,
          openai_file.testeract-baseline-file-2.id,
          openai_file.testeract-baseline-file-3.id
        ]
      }
    }
  }
```

**Comment by nathan-gilbert** [comment]
Also fails with the same error when creating an assistant. I'm wondering if the OpenAI Teams API don't have the `tool_resources` parameter yet?

**Comment by skyscrapr** [comment]
No this is my fault. 
Due to it being a beta endpoint it requires a special header. 

I'm updating the tests to include this attribute.  

ETA 1 day.

**Comment by skyscrapr** [comment]
ok! firstly apologies. Got a bit lazy with the tests. This time I added your exact scenario to the tests so we should be more successful. Just released v1.2.2 with the fixes. Take a look and let me know.
